pysrc/scripts/evaluation.py

```diff
@@ -39,72 +42,167 @@ def create_evaluation_csv(runs: MetanomeRunBatch, output_file: str, config: Glob
+    df_grouped = dataframe.groupby(groupby_attr)
+    many_plots = False
+    if len(df_grouped) > 10:
+        many_plots = True
     
-    df_rate = df.groupby('sampling_rate')
     d = []
-    for rate, frame in df_rate:
-        for i in range(int(frame['tp'].mean())):
-            d.append([rate,'tp', 1])
-        for i in range(int(frame['fp'].mean())):
-            d.append([rate,'fp', 1])
-        # for i in range(int(frame['fn'].mean())):
-        #     d.append([rate,'fn', 1])
+    for group_identifier, frame in df_grouped:
+        d.append([group_identifier, frame['precision'].mean(), frame['recall'].mean(), frame['f1'].mean()])            
+    df_grouped = pd.DataFrame(data=d, columns=[groupby_attr, 'Precision', 'Recall', 'F1-Score']).set_index(groupby_attr)
 
-    df_rate = pd.DataFrame(d, columns=['rate', 'type', 'count'])
+    sns.lineplot(data=df_grouped, ax=axes)
     
-    sns.histplot(
-        df_rate,
-        x='rate',
-        hue='type',
-        hue_order=['tp', 'fp'],
-        multiple='stack',
-        ax=ax2,
-        linewidth=.3,
-    )
+    if many_plots: # If there are too many entries, attempt to make it somewhat readable...
+        axes.tick_params(axis='x', rotation=90)
+        axes.tick_params(axis='x', labelsize=4)
+    axes.set_xlabel(f"{groupby_attr}")
+    axes.set_ylabel("Percentages")
     
-    ax2.tick_params(axis='x', rotation=90)
-    ax2.tick_params(axis='x', labelsize=4)
-    ax2.set_xlabel("Sampling Rates")
-    ax2.set_ylabel("Count")
+    return axes
+
+
+def create_plot(dataframe: pd.DataFrame, groupby_attrs: list[str], plot_method: Callable, plot_folder: str, plot_fname: str, figsize = (15,10)):
+    f, axes = plt.subplots(1, len(groupby_attrs), figsize=figsize)
+    sns.despine(f)
     
-    plot_fname = f'plots_{config.arity}_{config.now.year}{config.now.month:02d}{config.now.day:02d}_{config.now.hour:02d}{config.now.minute:02d}{config.now.second:02d}.jpg' 
+    if len(groupby_attrs) > 1:
+        for ax, groupby_attr in zip(axes, groupby_attrs):
+            ax = plot_method(axes = ax, dataframe = dataframe, groupby_attr = groupby_attr)
+    else:
+        axes = plot_method(axes = axes, dataframe = dataframe, groupby_attr = groupby_attrs[0])
     
     plot_path = os.path.join(os.getcwd(), plot_folder, plot_fname)
     f.savefig(plot_path)
     return plot_path
 
 
+## For n-ary INDs, create plots showing TP,FP,FN per arity
+# TODO: IMPLEMENT
+def create_onion_plot(axes: matplotlib.axes.Axes, dataframe: pd.DataFrame, groupby_attr: str) -> str:
+    df_grouped = dataframe.groupby(groupby_attr)
+    d = []
+    for group_identifier, frame in df_grouped:
+        print(group_identifier)
+        tp, fp, fn = frame['tp'].str.split('; '), frame['fp'].str.split('; '), frame['fn'].str.split('; ')
+        
+        tps: Dict[int, list[int]] = {}; fps: Dict[int, list[int]] = {}; fns: Dict[int, list[int]] = {}
```

Comment:
```suggestion
        tps: dict[int, list[int]] = {}; fps: dict[int, list[int]] = {}; fns: dict[int, list[int]] = {}
```
And remove the import for `Dict`.